src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4460,8 +4460,6 @@ void StorageReplicatedMergeTree::assertNotReadonly() const
 SinkToStoragePtr StorageReplicatedMergeTree::write(const ASTPtr & /*query*/, const StorageMetadataPtr & metadata_snapshot, ContextPtr local_context)
 {
     const auto storage_settings_ptr = getSettings();
-    assertNotReadonly();
```

Comment:
I think you're right. It looks like some leftover.

Reply:
Well, I forgot, but this change was done on purpose. We allow creating `ReplicatedMergeTreeSink` even for table in read-only mode since the sink can now handle it with retries.

Currently, integration test for retries is failing because of this check